src/datreant/core/collections.py

```diff
@@ -148,32 +148,34 @@ def __getitem__(self, index):
         select out members.
 
         """
+        memberlist = self._list()
```

Comment:
One thing I did notice is that this `__getitem__` method is repeated for `Bundle` and `View`, which isn't the prettiest (and isn't nice for coverage etc).  Would it be worth refactoring this to inherit from some sort of `dtr_container` base class which defines the container-like behaviour in datreant?  There's a few minor quirks with uuid/str getitem, but you could still do something like:

```python
def __getitem__(self, index):
    # do my quirks
    if isinstance(index, string_types):
        # do uuid match, because Bundle does this as a special thing
    else:
        # defer to default Container-like behaviour
        super(Bundle, self).__getitem__(index)
```